auto publish to pypi on tagged commit

PolarBean opened the pull request:
After addressing the issues in the previous PR we are now using the same action as other brainglobe projects. The main change is I have set this repo to be a trusted publisher so we no longer require the twine api key from pypi.

PolarBean committed:
remove testing file

PolarBean committed:
auto publish to pypi on release

PolarBean committed:
update yml file and python versions

PolarBean committed:
remove scheduled deploy

PolarBean committed:
remove twine api key as we can instead set the repo to be a trusted publisher for token less pushing

PolarBean committed:
run black linter

PolarBean committed:
enforce line lengths

PolarBean committed:
enforce 79 char line length

PolarBean committed:
exclude files that will not be included in pypi

PolarBean committed:
remove line length from ruff as its failing test (its still specified elsewhere)

PolarBean committed:
fix issue with test

PolarBean committed:
update license file to avoid spdx warning

PolarBean committed:
ruff fixes

PolarBean committed:
correct line incorrectly removed from ingestion script and caught by precommit

PolarBean committed:
remove unused variable caught by precommit

PolarBean committed:
make ruff happy

PolarBean committed:
ruff reformat

PolarBean committed:
exclude utilities

PolarBean committed:
precommit passed locally but failed remotely, updating reqs and license

PolarBean committed:
move docstring to beginning of file

PolarBean committed:
tests failing bc of triple quoted comments

PolarBean committed:
ruff reformat

PolarBean committed:
test off by 0.5 voxels...

PolarBean committed:
rename test case

PolarBean commented:
this is ready for review. I believe the remote tests will fail as they are too heavy duty for a github action however I have run them locally and they are passing

PolarBean committed:
revert test value

PolarBean committed:
fix empty last line

PolarBean committed:
autofix files precommit

IgorTatarnikov commented on .github/workflows/test_and_deploy.yml:
```diff
@@ -0,0 +1,63 @@
+          - os: macos-latest
+            python-version: "3.11"
+          - os: windows-latest
+            python-version: "3.11"
```
We typically have these set to `3.13` (or the latest compatible version of Python being tested)

IgorTatarnikov commented on PyPiDeploy.md:
Is this file necessary? The instructions can be found https://brainglobe.info/community/developers/new_releases.html

IgorTatarnikov reviewed (commented):
Just a few comments regarding consistency/standardisation throughout. Feel free to leave those as separate issues that can be fixed in future PRs.

I would also suggest changing the imports for `brainglobe-ccf-translator` to be [absolute as opposed to relative](https://peps.python.org/pep-0008/#imports) throughout. At this point it's used interchangeably and standardising it would be nice!

PolarBean committed:
Auto detect same space atlases (#26)

* implement atlas synonyms to be more compatible with brainglobe

* return atlas synonyms among near misses

* Update README with atlas synonyms

Added information about brainglobe atlas API names and their conversions.

* precommit

* precommit and add documentation about running just one test

* add to ignore words list so that codespell doesnt think the method name is a typo

* include demba api name in autoconvert list

* update metadata and ingestion scripts to include dorr

* update readme

* run graph update script and update readme

* run graph update script and update readme

* update dorr age to p84

* update accessibility text as we are no longer using an image

* update ruff formatting

* run precommit

* precommit

* replace click with print

* add mode arg to suppress future warning

* ruff reformat

* unittest worked but pytest failed, here we introduce a fix for this

* drop workflows for 3.9 and 3.10 as they seem to run out of resources

* use atlas api to access atlas shapes

* remove license to make pre-commit happy

* Dorr mri (#27)

* include demba api name in autoconvert list

* update metadata and ingestion scripts to include dorr

* update readme

* run graph update script and update readme

* run graph update script and update readme

* update dorr age to p84

* update accessibility text as we are no longer using an image

* update ruff formatting

* run precommit

* precommit

* replace click with print

* add mode arg to suppress future warning

* ruff reformat

* unittest worked but pytest failed, here we introduce a fix for this

* drop workflows for 3.9 and 3.10 as they seem to run out of resources

* use atlas api to access atlas shapes

* remove license to make pre-commit happy

* Remove unnecessary whitespace in config.py

---------

Co-authored-by: Igor Tatarnikov <61896994+IgorTatarnikov@users.noreply.github.com>

PolarBean commented:
Hey @IgorTatarnikov I was just reviewing past commits and I release I somehow merged this without addressing your comments. So sorry, this was totally unintentional and I just realised!!!

I'll change the repo settings so that merging requires an approving review and make an issue to address the feedback in this review first thing.
